src/tests/suite/daos_mgmt.c

```diff
@@ -440,6 +440,50 @@ get_sys_info_test(void **state)
 	daos_mgmt_put_sys_info(info);
 }
 
+/*
+ * Each engine shall disconnect all local connections when destroying a pool.
+ */
+static void
+pool_destroy_disconnect_all(void **state)
+{
+	test_arg_t   *arg = *state;
+	uuid_t        uuid;
+	char	      uuid_str[37];
+	int           rc;
+
+	FAULT_INJECTION_REQUIRED();
+
+	if (arg->myrank != 0)
+		return;
+
+	print_message("creating pool synchronously ... ");
+	rc = dmg_pool_create(dmg_config_file, geteuid(), getegid(), arg->group, NULL /* tgts */,
+			     256 * 1024 * 1024 /* minimal size */, 0 /* nvme size */,
+			     NULL /* prop */, arg->pool.svc, uuid);
+	assert_rc_equal(rc, 0);
+	print_message("success uuid = "DF_UUIDF"\n", DP_UUID(uuid));
+
+	uuid_unparse_lower(uuid, uuid_str);
+	rc = daos_pool_connect(uuid_str, arg->group, DAOS_PC_RW, &arg->pool.poh,
+			       NULL /* pool info */, NULL /* ev */);
+
+	rc = daos_debug_set_params(arg->group, -1, DMG_KEY_FAIL_LOC,
```

Comment:
LGTM.

For potential future expanded tests I wonder about another potential test
- pool connect fails after pool_connect_iv_dist() potentially leaving stranded ds_pool_tgt_hdl on storage engines (and leaving no record of the handle in the PS in rdb I believe is what we had seen in testing). Then during subsequent destroy, it is left to pool_tgt_disconnect_all() to clean up, else strand storage.

Reply:
@kccain, that indeed sounds like a more realistic test than the one I came up with here. Thanks.